src/Function/Metric/Definitions.agda

```diff
@@ -0,0 +1,60 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Definitions of properties over distance functions
+------------------------------------------------------------------------
+
+-- The contents of this module should be accessed via `Function.Metric`.
+
+{-# OPTIONS --without-K --safe #-}
+
+module Function.Metric.Definitions where
+
+open import Algebra.Core using (Op₂)
+open import Data.Product using (∃)
+open import Function.Metric.Core using (DistanceFunction)
+open import Level using (Level)
+open import Relation.Binary.Core using (Rel; _Preserves₂_⟶_⟶_)
+open import Relation.Nullary using (¬_)
+
+private
+  variable
+    a i ℓ ℓ₁ ℓ₂ : Level
+    A : Set a
+    I : Set i
+
+-----------------------------------------------------------------------
+-- Properties
+
+Congruent : Rel A ℓ₁ → Rel I ℓ₂ → DistanceFunction A I → Set _
+Congruent _≈ₐ_ _≈ᵢ_ d = d Preserves₂ _≈ₐ_ ⟶ _≈ₐ_ ⟶ _≈ᵢ_
+
+Indiscernable : Rel A ℓ₁ → Rel I ℓ₂ → DistanceFunction A I → I → Set _
+Indiscernable _≈ₐ_ _≈ᵢ_ d 0# = ∀ {x y} → d x y ≈ᵢ 0# → x ≈ₐ y
+
+PositiveDefinite : Rel A ℓ₁ → Rel I ℓ₂ → DistanceFunction A I → I → Set _
+PositiveDefinite _≈ₐ_ _≈ᵢ_ d 0# = ∀ {x y} → x ≈ₐ y → d x y ≈ᵢ 0#
```

Comment:
Are you sure that's enough for `PositiveDefinite`? Don't you also need that if they are not equal then the distance is not 0 too?

Reply:
Yup, nice spot! Renamed this one to `Definite` and added `NonNegative` as well which is then used in `IsProtoMetric`.